File: Crio/crio_programming_interview_problems-master/KthLargestElementInAnArray/KthLargestElementInAnArray.py

def findKthLargest(nums, k):
    # CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    import heapq
    return heapq.nlargest(k, nums)[-1]
    # CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

if __name__ == '__main__':
    n = int(input())
    nums = input().split()
    nums = [int(i) for i in nums]
    k = int(input())
    result = findKthLargest(nums, k)
    print(result)
    

File: Crio/StackAndQueue/10_KthLargestElementInAnArray/KthLargestElementInAnArray.cpp

#include <bits/stdc++.h>
using namespace std;

class KthLargestElementInAnArray {
public:
    int findKthLargest(vector<int>& nums, int k) {
	// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        int n = nums.size();
        priority_queue<int  > pq;
        for ( int i = 0; i < n; i++) {
            pq.push(nums[i]);
        }
        int answer = 0;
        while (k > 0) {
            k--;
            answer = pq.top();
            pq.pop();
        }
        return answer;
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    }
};

int main() {
    int n, k;
    cin >> n;
    vector<int> nums(n);
    for ( int i = 0; i < n; i++) cin >> nums[i];
    cin >> k;
    int result = KthLargestElementInAnArray().findKthLargest(nums, k);
    cout << result;
    return 0;
}
